src/traces/scattergl/convert.js

```diff
@@ -393,17 +393,74 @@ function convertLinePositions(gd, trace, positions) {
         if(trace.line.shape === 'hv') {
             linePositions = [];
             for(i = 0; i < count - 1; i++) {
-                if(isNaN(positions[i * 2]) || isNaN(positions[i * 2 + 1])) {
+                if(isNaN(positions[i * 2]) || isNaN(positions[i * 2 + 1]) || isNaN(positions[i * 2 + 2]) || isNaN(positions[i * 2 + 3])) {
+                    if(!isNaN(positions[i * 2]) && !isNaN(positions[i * 2 + 1])) {
+                        linePositions.push(positions[i * 2]);
+                        linePositions.push(positions[i * 2 + 1]);
+                    } else {
+                        linePositions.push(NaN);
+                        linePositions.push(NaN);
+                    }
```

Comment:
I'm not quite sure what impact this change had, but I'm curious, why is it needed here and not in `'vh'` below?

Reply:
BTW it's not new in this PR, but `linePositions.push(NaN); linePosition.push(NaN);` is equivalent to `linePositions.push(NaN, NaN);` right? Could probably shorten this whole section quite a bit that way...

Reply:
@alexcjohnson you are right something was wrong .

@etpinard can you cherry pick this commit 

https://github.com/plotly/plotly.js/commit/cab8a06a9fd819a159f16cb5f796c36b5c9c644d

Reply:
> can you cherry pick this commit

git-applied in d8297a0

>  Could probably shorten this whole section quite a bit that way...

done in d708132